Minor revisions (#2)
bemasc/svcb-dane: draft-rebs-dnsop-svcb-dane.md

This draft applies the same logic to SVCB-compatible records.  Specifically, if SVCB resolution was entirely secure (including any AliasMode records and/or CNAMEs), then for each connection attempt derived from a SVCB-compatible record,

> * The TLSA base domain MUST be the final SVCB TargetName after following AliasMode indirection if applicable.

Reviewer: ```suggestion
* The initial TLSA base domain MUST be the final SVCB TargetName used for this connection attempt.  (Names appearing earlier in a resolution chain are not used.)
```

* The transport prefix MUST be the transport of this connection attempt (possibly influenced by the "alpn" SvcParam).
* The port prefix MUST be the port number of this connection attempt (possibly influenced by the "port" SvcParam).

> If the SVCB TargetName contains a CNAME record, clients MUST first try to use the CNAME target as the TLSA base domain then fallback to the final SVCB TargetName as the TLSA base domain, as described in {{Section 7 of RFC7671}}.

Reviewer: ```suggestion
If the SVCB TargetName contains a CNAME record, clients MUST first try to use the CNAME target as the TLSA base domain, with fallback to the final SVCB TargetName as the TLSA base domain, as described in {{Section 7 of RFC7671}}.
```

## HTTPS AliasMode

Given service URI `https://api.example.com` and records:

~~~
api.example.com.     HTTPS 0 svc4.example.net.

> svc4.example.net.    HTTPS 1 xyz.example-cdn.com.

Reviewer: I liked this better the other way.  HTTPS does allow the chain to terminate with an AliasMode record.  This way seems too similar to the previous example.

Reviewer 2: Reverted

xyz.example-cdn.com. A     192.0.2.1
~~~

The TLSA QNAME is `_443._tcp.xyz.example-cdn.com`.

## QUIC and CNAME

Given service URI `https://api.example.com` and records:

~~~
www.example.com.  CNAME api.example.com.

> api.example.com.  HTTPS 1 svc4.example.net alpn=h3 port=8443

Reviewer: ```suggestion
api.example.com.  HTTPS 1 svc4.example.net alpn=h2,h3 port=8443
```

svc4.example.net. CNAME xyz.example-cdn.com.
~~~

If the client supports `alpn=h3`, the transport label is set to `_quic` otherwise `_tcp` is used.

> The TLSA QNAME would be one of:

Reviewer: ```suggestion
The initial TLSA QNAME would be one of:
```

* `_8443._quic.svc4.example.net`
* `_8443._tcp.svc4.example.net`

If no TLSA record is found, the fallback TLSA QNAME would be one of:

* `_8443._quic.xyz.example-cdn.com`
* `_8443._tcp.xyz.example-cdn.com`

## New scheme ServiceMode

Given service URI `foo://api.example.com:8443` and record:

~~~
_8443._foo.api.example.com. SVCB 1 api.example.com.
~~~

> The TLSA QNAME is `_8443._$PROTO.api.example.com`, where $PROTO is the appropriate value as discussed in {{protocols}} matching the client-selected transport.

Reviewer: ```suggestion
The TLSA QNAME is `_8443._$PROTO.api.example.com`, where $PROTO is the appropriate value for the client-selected transport as discussed in {{protocols}} .
```
